dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java

```diff
@@ -1085,6 +1090,12 @@ public EventSearchParams getFromUrl( String program, String programStage, Progra
                 params.getDataElements().add( dataElement );
             }
         }
+        
+        if ( assignedUserSelectionMode != null && assignedUserSelectionMode.equals( AssignedUserSelectionMode.CURRENT )
```

Comment:
AssignedUserSelectionMode.CURRENT.equals( assignedUserSelectionMode )

Reply:
I wanted to add the flexibility of making AssignedUserSelectionMode not mandatory if "assignedUserIds" are explicitly provided. In that case, changing the condition to what you said would fail. For example the api should work if "auMode=Provided&au=1,2,3" as well as simply "au=1,2,3"